File: web/src/app/models/resourceType.ts
/**
 * This class represents the supported  types of resources.
 * It converts a between a string representation of a resource type and an object containing value, viewValue, and pathDescriptor.
 * @author Kyanoush Yahosseini
 *
 */
export class ResourceType {
  value: string;
  viewValue: string;
  pathDescriptor: string;

  constructor(value: string | number) {
    switch (value) {
      case 'SAN_OU':
        return {value: 'SAN_OU', viewValue: 'Gruppenlaufwerk (S:\\OE)', pathDescriptor: 'Pfad:'};
      case 'SAN_PROJECT':
        return {value: 'SAN_PROJECT', viewValue: 'Projektlaufwerk (S:\\Projekte)', pathDescriptor: 'Pfad:'};
      case 'SAN_DATA':
        return {value: 'SAN_DATA', viewValue: 'WissDaten (S:\\Wissdaten)', pathDescriptor: 'Pfad:'};
      case 'OPENBIS':
        return {value: 'OPENBIS', viewValue: 'OpenBIS', pathDescriptor: 'ID:'};
      case 'GIT':
        return {value: 'GIT', viewValue: 'Git Repositorium', pathDescriptor: 'URL:'};
      case 'DOI':
        return {value: 'DOI', viewValue: 'DOI Referenz', pathDescriptor: 'DOI:'};
      case 'DMS':
        return {value: 'DMS', viewValue: 'DMS/VBS', pathDescriptor: 'Adresse:'};
      default:
        return ({value: null, viewValue: null, pathDescriptor: null});

    }
  }
}

File: server/src/main/java/de/rki/mamodar/ResourceType.java
package de.rki.mamodar;

/**
 * The enum of all possible resource types.
 * @author Kyanoush Yahosseini
 */
public enum ResourceType {
  /**
   * SAN resource type.
   */
  SAN("WissData"),
  /**
   * LOCAL resource type.
   */
  LOCAL("Lokal"),
  /**
   * OPENBIS resource type.
   */
  OPENBIS("OpenBIS"),
  /**
   * GIT resource type.
   */
  GIT("git"),
  /**
   * DOI resource type.
   */
  DOI("doi"),
  /**
   * PRIV resource type.
   */
  PRIV("Personal"),
  /**
   * COUD resource type.
   */
  CLOUD("Cloud");


  private final String representation;

  ResourceType(String representation) {
    this.representation = representation;
  }

}
